bcgov/mds
# [MDS-4568] Major Mine Application Step 3: Review and Submit - MineSpace

## Objective 

[MDS-4568](https://bcmines.atlassian.net/browse/MDS-4568)

_What Changed?_

- Create new route component MajorMineApplicationReviewSubmit(Step 3)
- Port over DocumentLink and DocumentViewer from Core-Web
- Integrate DocumentLink & DocumentViewer into Step 3 DocumentTable
- Add Major Mine Application document deletion support on Step 3 Document Table
- Navigating from ProjectOverviewTab via tab/link will direct to entry page(Start/Resume states)
- On successful submission(Step 3) user is redirected to success/confirmation page
- On final submission email is sent to project contacts with Major Mine Application link and submitted document list
- Various propType/React warning fixes
- Navigating from Step 2 -> 3 will automatically save draft

_How to Test:_

- Navigate from ProjectOverviewTab to Major Mine Application(MMA) tab or stage link:
    - If MMA has not been created(no draft or submission) the entry tab will show "Start" instructions and link
    - If MMA has already been created the entry tab will show "Resume" instructions and link
        - Verify clicking Start or Resume directs to Step 1/3 respectively: 
![image](https://user-images.githubusercontent.com/13891487/183219842-af81d757-df61-4ca8-860c-d3db487b46b7.png)
![image](https://user-images.githubusercontent.com/13891487/183219786-e73ec080-307c-4c71-92ac-1a182205586e.png)

- Navigate to Step 3
    - Verify layout and data matches designs
    - Verify "Submit Now" button is disabled until confirmation checkbox at bottom of page is checked: 
![image](https://user-images.githubusercontent.com/13891487/183219942-e037d46e-7b5e-444c-8ca7-45c15b2eec7a.png)
    - Verify you can download/delete non PDF files on DocumentTable
    - Verify you can download/preview/delete PDF files on DocumentTable(download seems to open preview screen, may be an existing bug)
 - Click "Submit Now" and upon success of submission be directed to confirmation/success page
     - Verify Success page matches designs
     - Verify attached links direct to correct destination: 
![image](https://user-images.githubusercontent.com/13891487/183220062-85d33878-bb56-43b2-9663-d674f62b30c3.png)
